main.py

```diff
@@ -1,209 +1,230 @@
+        print_cursor_value(val, "D")
+        return 0
     return -1
+
+
+def print_cursor_value(val, letter):
+    unicode_prefix = u"\u001b["
+    print(f'{unicode_prefix}{val}{letter}', end='')
+
+
 def setColor(val):
-    if(isinstance(val, str)):
-        val=val.upper()
-        if(val=="BLACK"):
-            print("\u001b[30m",end="")
-        elif(val=="RED"):
-            print("\u001b[31m",end="")
-        elif(val=="GREEN"):
-            print("\u001b[32m",end="")
-        elif(val=="YELLOW"):
-            print("\u001b[33m",end="")
-        elif(val=="BLUE"):
-            print("\u001b[34m",end="")
-        elif(val=="MAGENTA"):
-            print("\u001b[35m",end="")
-        elif(val=="CYAN"):
-            print("\u001b[36m",end="")
-        elif(val=="WHITE"):
-            print("\u001b[37m",end="")
-        elif(val=="RESET"):
-            print("\u001b[0m",end="")
-        else:
-            return -1
-        return 0
+    if isinstance(val, str):
+        val = val.upper()
+        colors = {"BLACK": "\u001b[30m",
+                  "RED": "\u001b[31m",
+                  "GREEN": "\u001b[32m",
+                  "YELLOW": "\u001b[33m",
+                  "BLUE": "\u001b[34m",
+                  "MAGENTA": "\u001b[35m",
+                  "CYAN": "\u001b[36m",
+                  "WHITE": "\u001b[37m",
+                  "RESET": "\u001b[0m"}
+        if val in colors.keys():
+            print(colors[val], end="")
+            return 0
     return -1
+
+
 def printList(arr):
-    for i in range(len(arr)):
-        print("  " + arr[i],end="\n\r")
+    for word in arr:
+        print(f"  {word}", end="\n\r")
+
+
 def hideCursor():
-    print("\u001b[?25l",end='')
+    print("\u001b[?25l", end='')
+
+
 def resetCursor():
-    print("\u001b[?0l",end='')
-def singleChoice(arr,cursorColor="Blue",textOrVal="Val"):
-    #save terminal settings
+    print("\u001b[?0l", end='')
+
+
+def singleChoice(arr, cursorColor="Blue", textOrVal="Val"):
+    # save terminal settings
     fd = sys.stdin.fileno()
     old_settings = termios.tcgetattr(fd)
-    #set terminal to raw mode
+    # set terminal to raw mode
     tty.setraw(sys.stdin)
-    #read output
-    charIn=0
+    # read output
+    charIn = 0
     printList(arr)
     hideCursor()
     cursorUp(len(arr))
-    position=0
+    position = 0
     setColor(cursorColor)
-    print(">",end="")
+    print(">", end="")
     setColor("Reset")
     cursorLeft(1)
-    while charIn!=13: #whilst enter not pressed
+    enter_not_pressed = charIn != 13
+    while enter_not_pressed:
         sys.stdout.flush()
-        charIn=ord(sys.stdin.read(1))#Get input
-        if(charIn==27):#ESC pressed
-            charIn=ord(sys.stdin.read(1))
-            if(charIn==91):#[ pressed
-                charIn=ord(sys.stdin.read(1))#Get input
-                if(charIn==65):#Up
-                    if(position>0):
-                        print(" ",end="")
+        charIn = get_input_char()  # Get input
```

Comment:
turn ord(sys.stdin.read(1)) into a function so you know what it does implicitly and don't need a comment